internal/cli/atlas/backup/compliancepolicy/copyprotection.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compliancepolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201003/admin"
+)
+
+type CopyProtectionOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	policy *atlasv2.DataProtectionSettings
+	store  store.CompliancePolicy
+	enable bool
+}
+
+const (
+	enable  string = "enable"
+	disable string = "disable"
+	active  string = "ACTIVE"
+)
+
+func (opts *CopyProtectionOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *CopyProtectionOpts) PreRun() error {
+	currentPolicy, err := opts.store.DescribeCompliancePolicy(opts.ConfigProjectID())
+	if err != nil {
+		return err
+	}
+	opts.policy = currentPolicy
+	return nil
+}
+
+func (opts *CopyProtectionOpts) copyProtectionWatcher() (bool, error) {
+	res, err := opts.store.DescribeCompliancePolicy(opts.ConfigProjectID())
+	if err != nil {
+		return false, err
+	}
+	if res.GetState() == "" {
+		return false, fmt.Errorf("could not access State field")
```

Comment:
if you don't need string interpolation 
```suggestion
		return false, errors.New("could not access State field")
```